### PR TITLE
Add _keys functions

### PR DESCRIPTION
**Pull request by mtmorgan**

### CODE REVIEW

File: R/AbstractAnnData.R [hunk 1]
```diff
@@ -68,6 +68,14 @@ AbstractAnnData <- R6::R6Class("AbstractAnnData",
...
       nrow(self$var)
     },
+    #' @description Keys ('column names') of `obs`.
+    obs_keys = function() {
+      .abstract_function()
+    },
+    #' @description Keys ('column names') of `var`.
+    var_keys = function() {
+      .abstract_function()
+    },
```
Comment (rcannood) [inline R/AbstractAnnData.R:78]: Can we not implement this in the abstract class?

```suggestion
    obs_keys = function() {
      names(self$obs)
    },
    #' @description Keys ('column names') of `var`.
    var_keys = function() {
      names(self$var)
    },
```

File: R/InMemoryAnnData.R [hunk 2]
```diff
@@ -162,6 +162,14 @@ InMemoryAnnData <- R6::R6Class("InMemoryAnnData",
...
       private$.X <- private$.validate_matrix(X, "X")
       private$.layers <- private$.validate_layers(layers)
+    },
+    #' @description `obs_keys()` returns the keys ('column names') of `obs`.
+    obs_keys = function() {
+      names(private$.obs)
+    },
+    #' @description `var_keys()` returns the keys ('column names') of `var`.
+    var_keys = function() {
+      names(private$.var)
```
Comment (rcannood) [inline R/InMemoryAnnData.R:172]: If the other suggestion is accepted.

```suggestion
```

**Review by rcannood — CHANGES_REQUESTED** [review]

File: R/AbstractAnnData.R (hunk 1 above)
Comment (mtmorgan) [reply R/AbstractAnnData.R:78]: Is it guaranteed (or 'expected') that self$var returns something with `names()`?

Comment (rcannood) [reply R/AbstractAnnData.R:78]: Indeed :)

**Review by rcannood — APPROVED** [review]

**Pull request merged**